lib/rules/no-promise-executor-return.js

```diff
@@ -59,6 +60,68 @@ function isPromiseExecutor(node, scope) {
         isGlobalReference(parent.callee, getOuterScope(scope));
 }
 
+/**
+ * Checks if the given node is a void expression.
+ * @param {ASTNode} node The node to check.
+ * @returns {boolean} - `true` if the node is a void expression
+ */
+function expressionIsVoid(node) {
+    return node.type === "UnaryExpression" && node.operator === "void";
+}
+
+/**
+ * Fixes the linting error by prepending "void " to the given node
+ * @param {Object} sourceCode context given by context.sourceCode
+ * @param {ASTNode} node The node to fix.
+ * @param {Object} fixer The fixer object provided by ESLint.
+ * @returns {Array<Object>|Object} - An array of fix objects or fix to apply to the node.
+ */
+function voidPrependFixer(sourceCode, node, fixer) {
+
+    /*
+     * prepending `void ` will fail if the node has a lower precedence than void
+     */
+    if (astUtils.getPrecedence(node) < astUtils.getPrecedence({ type: "UnaryExpression", operator: "void" })) {
+
+        // check if there are parentheses around the node to avoid redundant parentheses
+        if (!(sourceCode.getTokenBefore(node).value === "(" && sourceCode.getTokenAfter(node).value === ")")) {
```

Comment:
We can use [`astUtils.isParenthesised`](https://github.com/eslint/eslint/blob/840a26462bbf6c27c52c01b85ee2018062157951/lib/rules/utils/ast-utils.js#L514) helper function here.